lib/wallet.js

```diff
@@ -315,6 +315,144 @@ class WalletClient extends Client {
     return this.get(`/wallet/${id}/coin/${hash}/${index}`);
   }
 
+  /**
+   * Get name state for the given name.
+   * {@see hsd.NameState}
+   * @param {String} id
+   * @param {String} name
+   * @returns {Promise}
+   */
+
+  getName(id, name) {
+    return this.get(`/wallet/${id}/name/${name}`);
+  }
+
+  /**
+   * Get name state for all names
+   * that the wallet is managing.
+   * {@see hsd.NameState}
+   * @param {String} id
+   * @param {Object} options
+   * @returns {Promise}
+   */
+
+  getNames(id, options) {
+    return this.get(`/wallet/${id}/name`, options);
+  }
+
+  /**
+   * Get bids, reveals and name state
+   * for the given name.
+   * {@see hsd.NameState}
+   * {@see hsd.BlindBid}
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {String} name
+   * @returns {Promise}
+   */
+
+  getAuctionByName(id, name) {
+    return this.get(`/wallet/${id}/auction/${name}`);
+  }
+
+  /**
+   * Get bids, reveals and name state
+   * for all names the wallet manages.
+   * {@see hsd.NameState}
+   * {@see hsd.BlindBid}
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {Object} options
+   * @returns {Promise}
+   */
+
+  getAuctions(id, options) {
+    return this.get(`/wallet/${id}/auction`, options);
+  }
+
+  /**
+   * Get wallet bids for a given name.
+   * {@see hsd.BlindBid}
+   * @param {String} id
+   * @param {String?} name
+   * @param {Object?} options
+   * @param {Boolean?} options.own
+   * @returns {Promise}
+   */
+
+  getBidsByName(id, name, options) {
+    return this.get(`/wallet/${id}/bid/${name}`, options);
+  }
+
+  /**
+   * Get wallet bids for all names
+   * the wallet manages.
+   * {@see hsd.BlindBid}
+   * @param {String} id
+   * @param {Object?} options
+   * @param {Boolean?} options.own
+   * @returns {Promise}
+   */
+
+  getBids(id, options) {
+    return this.get(`/wallet/${id}/bid`, options);
+  }
+
+  /**
+   * Get wallet reveal for a given name.
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {String?} name
+   * @param {Object?} options
+   * @param {Boolean?} options.own
+   * @returns {Promise}
+   */
+
+  getRevealsByName(id, name, options) {
+    return this.get(`/wallet/${id}/reveal/${name}`, options);
+  }
+
+  /**
+   * Get wallet reveals for all names
+   * the wallet manages.
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {Object?} options
+   * @param {Boolean?} options.own
+   * @returns {Promise}
+   */
+
+  getReveals(id, options) {
+    return this.get(`/wallet/${id}/reveal`, options);
+  }
+
+  /**
+   * Get name resource.
+   * {@see hsd.Resource}
+   * @param {String} id
+   * @param {String} name
+   * @param {Object?} options
+   * @returns {Promise}
+   */
+
+  getResource(id, name, options) {
```

Comment:
I do not think this endpoint accepts options anymore.

Reply:
I added `options` here to future proof the method, so that they could be added into `hsd` without having to update this codebase. I can remove it if you think that its premature

Reply:
I think that prematurely providing the options hash can be confusing to library consumers. We do not get a considerable advantage from including it now, but it sets a bad precedent for including dead code.